packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -421,15 +421,29 @@ function transform(codeGen: CodeGen): t.Expression {
                 } else if (name === 'style') {
                     // Handle style attribute:
                     // - expression values are turned into a `style` property.
-                    // - string values are parsed and turned into a `styleMap` object associating
-                    //   each property name with its value.
+                    // - string values are parsed and turned into a `styles` array
+                    // containing triples of [name, value, important (optional)]
                     if (value.type === IRAttributeType.Expression) {
                         const styleExpression = bindExpression(value.value, element);
                         data.push(t.property(t.identifier('style'), styleExpression));
                     } else if (value.type === IRAttributeType.String) {
                         const styleMap = parseStyleText(value.value);
-                        const styleObj = objectToAST(styleMap, (key) => t.literal(styleMap[key]));
-                        data.push(t.property(t.identifier('styleMap'), styleObj));
+                        const styles: Array<[string, string] | [string, string, boolean]> =
```

Comment:
Would this be better off as a helper function in `helpers.ts`?

At first I wanted to ask why this wasn't just added to `parseStyleText` from line 430, but I think it still makes sense to keep that functionality separate.

Reply:
Done.